src/Microsoft.Azure.EventHubs/Primitives/ClientEntity.cs

```diff
@@ -55,6 +65,43 @@ public RetryPolicy RetryPolicy
         /// <returns>The asynchronous operation</returns>
         public abstract Task CloseAsync();
 
+        /// <summary>
+        /// Registers a <see cref="EventHubsPlugin"/> to be used with this client.
+        /// </summary>
+        public virtual void RegisterPlugin(EventHubsPlugin eventHubPlugin)
```

Comment:
>eventHub [](start = 59, length = 8)

Can you make sure it is 'eventhubs' everywhere since this needs to be same as the service name. 'Event Hub' refers to single entity.